tests/test_node.py

```diff
@@ -90,18 +177,49 @@ def test_popen_as_root(
             generate_config_path=oss_artifact,
         ) as cluster:
             (master, ) = cluster.masters
-            popen_1 = master.popen_as_root(
+
+            # The user is configurable.
+            # Create a user.
+            adduser_result = master.run(
+                args=['adduser', 'testuser'], user='root'
+            )
+            assert adduser_result.returncode == 0
+            # Prepare the user account for public key SSH access from the test
+            mkdir_result = master.run(
+                args=['mkdir', '-p', '/home/testuser'], user='root'
+            )
+            assert mkdir_result.returncode == 0
+            cp_result = master.run(
+                args=['cp', '-a', '/root/.ssh', '/home/testuser/.ssh'],
+                user='root'
+            )
+            assert cp_result.returncode == 0
+            chown_result = master.run(
+                args=['chown', '-R', 'testuser', '/home/testuser/.ssh'],
+                user='root'
+            )
+            assert chown_result.returncode == 0
+            # Remove stray file that prevents non-root SSH.
+            # https://ubuntuforums.org/showthread.php?t=2327330
+            rm_result = master.run(
+                args=['rm', '-f', '/run/nologin'], user='root'
```

Comment:
@timaa2k Do you agree that this should be done automatically? Maybe at the `cluster` level, or maybe at the DC/OS Docker backend level.

Reply:
I think it should be done at the specific backend level. dcos-e2e should require this for every potential backend in order to guarantee that it provides the run as user functionality.